cpp/src/lance/arrow/type.cc

```diff
@@ -75,8 +76,13 @@ ::arrow::Result<std::string> ToLogicalType(std::shared_ptr<::arrow::DataType> dt
                        dict_type->value_type()->ToString(),
                        dict_type->index_type()->ToString(),
                        dict_type->ordered());
+  } else if (::lance::arrow::is_extension(dtype)) {
+    // TODO what if extension_name has a `:` in it?
```

Comment:
@eddyxu wdyt about this? I guess it's not super urgent for now but when we tackle parametric extensions this might make parsing really complicated?

Reply:
Lets report error from the write path (here) first? 
like `if (extension_name.find(":") ...) return Error`. 

We can use `|` or something else as well? Or add a new Field.extension_name field?

Reply:
sg. i’m reporting an error here for now